pkg/controller/warehouses/warehouses.go

```diff
@@ -304,6 +378,12 @@ func (r *reconciler) syncWarehouse(
 				Reason:  "AwaitingFreightCreation",
 				Message: "Freight creation from latest artifacts is in progress",
 			},
+			&metav1.Condition{
+				Type:    kargoapi.ConditionTypeReady,
+				Status:  metav1.ConditionTrue,
+				Reason:  "ArtifactsDiscovered",
+				Message: conditions.Get(&status, kargoapi.ConditionTypeHealthy).Message,
+			},
```

Comment:
I don't understand this. You're setting Ready to false and then immediately setting it to true.